SwanCustomEnvironments/swancustomenvironments/serverextension.py

```diff
@@ -57,50 +50,24 @@ def _launch_makenv(self) -> subprocess.Popen:
         if nxcals:
             arguments.append("--nxcals")
 
-        return subprocess.Popen([self.makenv_path, *arguments], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-
-    async def _process_first_get(self, makenv_process) -> None:
-        """
-        Reads from the stdout of the makenv.sh process and sends it to the frontend.
-        This is done by the initial get request, which triggers the makenv.sh process.
-        """
-        SwanCustomEnvironmentsApiHandler.writing_log = True
-
-        # The first and any subsequent get requests will read from the process
-        # and stream out its output
         with open(self.LOG_FILE, "w") as log_file:
-            for line in iter(makenv_process.stdout.readline, b""):
-                line = line.decode("utf-8")
-                log_file.write(line)
+            # Create the environment asynchronously and send its output to the log file
+            return await asyncio.create_subprocess_exec(self.makenv_path, *arguments, stdout=log_file, stderr=log_file)
+
+    async def _process_log_stream(self, makenv_process: asyncio.subprocess.Process) -> None:
+        """Reads from the log file and streams output to the client."""
+        with open(self.LOG_FILE, "r") as log_file:
+            while True:
+                line = log_file.readline()
+                if not line and makenv_process.returncode is not None:
+                    break  # Process finished, no more logs to read
                 await self._send_line(line)
```

Comment:
Here you are sending an empty line in case no line can be read from the file but the returncode is still None (the makenv process is running).

Reply:
I see, you protect for that case in `_send_line`, nevermind.